### PR TITLE
optimizations/refactoring

### PR DESCRIPTION
**Pull request by Lagovas**
* added interface for Proxy that process mysql/postgresql connections and unified signatures for mysql/postgresql proxy methods to be compatible with one interface
* added interface for decryptors and unified signatures for db specific decryptors
* added decryptor/proxy factories that initialized at start of program and then used after accepting new connections. old code did initialization of components in different steps of processing connections (it was from our first version of server and need refactoring)
* initialize all db specific settings and components in main function once
* dropped redundant and unused fields/methods/constants in project (found them when tested golangci-lint : )
* refactored and encapsulated switching to tls in postgresql proxy (it was need to unifying usage of abstract proxy component without postgresql specific logic)
* optimized OnQueryObserver logic to re-use parsed queries between several observers (to avoid several parsing query, reuse ast tree and serialize to string only at end)
* add using absolute paths in integration tests to allow re-use in another projects/repos
* make some methods in our default keystore public to use in enterprise version

As I remember, I didn't change any public CLI or API interfaces that should be changed in documentation. And didn't change any behavior of server. Touched only some log messages (uppercase or change level from DEBUG to INFO)

### CODE REVIEW

File: decryptor/mysql/packet.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func (packet *MysqlPacket) replaceQuery(newQuery string) {
 
 // readPacket read header to struct and return payload as return result or error
 func (packet *MysqlPacket) readPacket(connection net.Conn) ([]byte, error) {
-	if _, err := connection.Read(packet.header); err != nil {
+	if _, err := io.ReadFull(connection, packet.header); err != nil {
```
Comment (vixentael) [inline decryptor/mysql/packet.go]: what is the difference?

File: decryptor/postgresql/proxy.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &proxyFactory{
+		dataEncryptor: dataEncryptor,
+		setting:       proxySetting,
+	}, nil
+}
+
+// New return mysql proxy implementation
```
Comment (vixentael) [inline decryptor/postgresql/proxy.go]:
```suggestion
// New return postgresql proxy implementation
```

File: keystore/filesystem/filenames.go [hunk 3]
```diff
@@ -18,11 +18,8 @@ package filesystem
 
 import (
 	"fmt"
-	"sync"
 )
 
-var lock = sync.RWMutex{}
```
Comment (vixentael) [inline keystore/filesystem/filenames.go]: bye-bye mutex 😢

File: decryptor/mysql/packet.go (hunk 1 above)
Comment (Lagovas) [reply decryptor/mysql/packet.go]: helper read more than once (if first read call was not full and return only part of data cached in os buffer). until some error or until no data was returned. ReadFull read in loop until read expected num of bytes. I meet one failed test when here read return part of packet and without error

File: keystore/filesystem/filenames.go (hunk 3 above)
Comment (Lagovas) [reply keystore/filesystem/filenames.go]: it wasn't used)

**Pull request merged**